src/backend/cgcod.c

```diff
@@ -726,7 +726,7 @@ code *prolog()
     assert(usedalloca != 1);
     AllocaOff = alignsection(usedalloca ? (Foff - REGSIZE) : Foff, REGSIZE, bias);
 
-    CSoff = alignsection(AllocaOff - cstop * REGSIZE, REGSIZE, bias);
+    CSoff = alignsection(AllocaOff - (int)cstop * REGSIZE, REGSIZE, bias);
```

Comment:
`AllocaOff` being an `int` seems to be the source of that problem. Working on a fix.